### DISCUSSION TITLE
Scalability

### DISCUSSION THREAD

**Discussion by kotyara85** [k8gb-io/k8gb#1724] [General]
👋 everyone,
Looking into the arch diagrams, it looks like basically every cluster will announce itself to the upstream DNS server.
What happens if you have a thousand clusters? 1000 NS records will get pushed to the upstream?

**Comment by abaguas** [comment]
Correct. This should not be a problem, the DNS infrastructure is robust and can handle it.
Each server will then query every other server to decide which records to publish. This operation will take some time, so a failover interval of 1s won't be feasible. But the same would apply with an external probing solution
